### PR TITLE
Replace tmpdir with tmp_path

### LINKED ISSUE #1039: TST: Replace tmpdir with tmp_path in pytest
`tmpdir` is considered legacy now and should be replaced with `tmp_path`.

https://docs.pytest.org/en/stable/how-to/tmp_path.html

### PR DESCRIPTION
**Pull request by rosteen**
Closes #1039 .

### CODE REVIEW

File: specutils/io/default_loaders/tests/test_jwst_reader.py [hunk 1]
```diff
@@ -366,7 +366,7 @@ def generate_s3d_wcs():
 
 
 @pytest.fixture()
-def tmp_asdf(tmpdir):
+def tmp_asdf(tmp_path):
```
Comment (pllim) [inline specutils/io/default_loaders/tests/test_jwst_reader.py:369]: But why? It looks unused.

```suggestion
def tmp_asdf():
```

**Review by pllim — COMMENTED** [review]
Seems uncontroversial enough though maybe one of the tests does not need `tmp_path` in the first place.

File: specutils/io/default_loaders/tests/test_jwst_reader.py (hunk 1 above)
Comment (rosteen) [reply specutils/io/default_loaders/tests/test_jwst_reader.py:369]: Hm, it does indeed 🤔. I'm surprised codestyle didn't catch that.

Comment (pllim) [reply specutils/io/default_loaders/tests/test_jwst_reader.py]: Some fixtures do magic behind the scenes but I think tmp_path is not needed unless it is explicitly used?

File: specutils/io/default_loaders/tests/test_jwst_reader.py [hunk 2]
```diff
@@ -366,7 +366,7 @@ def generate_s3d_wcs():
 
 
 @pytest.fixture()
-def tmp_asdf(tmpdir):
+def tmp_asdf():
```
Comment (pllim) [inline specutils/io/default_loaders/tests/test_jwst_reader.py:369]: What hurts my head is `tmp_asdf` itself is a fixture but it took in `tmpdir` fixture. What was it trying to do?

**Review by pllim — APPROVED** [review]
LGTM now as long as CI agrees with me. Thanks!

**Comment by pllim** [comment]
I will let you merge.

File: specutils/io/default_loaders/tests/test_jwst_reader.py (hunk 2 above)
Comment (rosteen) [reply specutils/io/default_loaders/tests/test_jwst_reader.py:369]: It is a mystery 😕

**Pull request merged**